tvb_framework/tvb/adapters/creators/siibra_base.py

```diff
@@ -0,0 +1,351 @@
+            parsed_ids.extend(ids_list_from_range)
+        else:
+            s_int = int(s)
+            parsed_ids.append(s_int)
+
+    # convert the subject ids list back to strings into the required format
+    parsed_ids_strings = [str(id).zfill(3) for id in parsed_ids]
+    return parsed_ids_strings
+
+
+def init_siibra_params(atlas_name, parcellation_name, subject_ids):
+    atlas = siibra.atlases[atlas_name] if atlas_name else None
+    parcellation = siibra.parcellations[parcellation_name] if parcellation_name else None
+
+    if atlas and parcellation:
+        compatible = check_atlas_parcellation_compatible(atlas, parcellation)
+        if not compatible:
+            LOGGER.error(f'Atlas {atlas.name} does not contain parcellation {parcellation.name}. '
+                         f'Please choose a different atlas and/or parcellation')
+            return
+
+    if atlas and not parcellation:
+        LOGGER.warning(f'No parcellation was provided, so a default one will be selected.')
+        parcellations = get_parcellations_for_atlas(atlas)
+        no_parcellations = len(parcellations)
+        if no_parcellations < 1:
+            LOGGER.error(f'No default parcellation was found for atlas {atlas.name}!')
+            return
+        if no_parcellations > 1:
+            LOGGER.info(
+                f'Multiple parcellations were found for atlas {atlas.name}. An arbitrary one will be selected.')
+        parcellation = parcellations[0]
+
+    if not atlas and parcellation:
+        LOGGER.warning('A parcellation was provided without an atlas, so a default atlas will be selected.')
+        atlases = get_atlases_for_parcellation(parcellation)
+        no_atlases = len(atlases)
+        if no_atlases < 1:
+            LOGGER.error(f'No default atlas containing parcellation {parcellation.name} was found!')
+            return
+        if no_atlases > 1:
+            LOGGER.info(
+                f'Multiple atlases containing parcellation {parcellation_name} were found. '
+                f'An arbitrary one will be selected')
+        atlas = atlases[0]
+
+    if not atlas and not parcellation:
+        LOGGER.warning(f'No atlas and no parcellation were provided, so default ones will be selected.')
+        atlas = siibra.atlases[DEFAULT_ATLAS]
+        parcellation = siibra.parcellations[DEFAULT_PARCELLATION]
+
+    LOGGER.info(f'Using atlas {atlas.name} and parcellation {parcellation.name}')
+
+    if not subject_ids:
+        LOGGER.info(
+            f'No list of subject ids was provided, so the connectivities will be computed for all available subjects!')
+        subject_ids = 'all'
+    elif subject_ids != 'all':
+        subject_ids = parse_subject_ids(subject_ids)
+
+    return atlas, parcellation, subject_ids
+
+
+# ######################################## COMMON CONNECTIVITY METHODS #################################################
+def get_connectivity_component(parcellation, component):
```

Comment:
Maybe for param `component` don't use these 3 hard-coded strings, but rather directly the siibra modality or a local enum ?